internal/pkg/remote/client_fips_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build requirefips
+
+package remote
+
+import (
+	_ "embed"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
+	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
+	"github.com/elastic/elastic-agent/pkg/core/logger/loggertest"
+)
+
+//go:embed testdata/rsa_1024.cert.pem
+var rsa1024CertPEM string
+
+//go:embed testdata/rsa_1024.key.pem
+var rsa1024KeyPem string
+
+func TestClientWithUnsupportedTLSConfig(t *testing.T) {
```

Comment:
After discussion with the FIPS working group today, it was decided that we don't want to validate configuration when it's loaded/created (see more: https://github.com/elastic/elastic-agent-libs/pull/311#issuecomment-2821694917).  Instead, we want to let the TLS handshake validate it.  So this test will need to be replaced by one that attempts to use the configuration in a TLS handshake against a FIPS-compliant server.